llvm/lib/Transforms/Utils/SimplifyCFG.cpp

```diff
@@ -6545,6 +6544,17 @@ static bool SwitchToLookupTable(SwitchInst *SI, IRBuilder<> &Builder,
 
   BranchInst *RangeCheckBranch = nullptr;
 
+  if (UseSwitchConditionAsTableIndex) {
```

Comment:
Please add a comment about what this does (grows the table to cover all possible index values to avoid the range check).

It would also be good if the commit message explained this a bit better: it's not really about mask operations, but any situation when the index range is limited. One way to do it is to include a small C example in the commit message.

Reply:
Thanks for the update!

For the commit message, I'd skip the godbolt link since that example is different. Also the text says it grows the table to the next pow2 value, which I don't think is true -- it grows it to the maximum index value, which doesn't have to be a power of two.

Reply:
Thanks,  I'll update the commit message.
 it grows it to the maximum index value, but finally it will grow the table to the next pow2 value.
`uint64_t TableSizePowOf2 = NextPowerOf2(std::max(7ULL, TableSize - 1ULL));`